src/Illuminate/Testing/Constraints/HasInDatabase.php

```diff
@@ -113,7 +113,7 @@ protected function getAdditionalInfo($table)
     public function toString($options = 0): string
     {
         foreach ($this->data as $key => $data) {
-            $output[$key] = $data instanceof Expression ? (string) $data : $data;
+            $output[$key] = $data instanceof Expression ? $data->getValue($this->database->getQueryGrammar()) : $data;
```

Comment:
Why not just 

```suggestion
            $output[$key] = $this->database->getQueryGrammar()->wrap($data);
```

https://github.com/laravel/framework/blob/3a46fa307c73c8371ea6c72129f7c23a2f52c7f3/src/Illuminate/Database/Grammar.php#L46-L74

Reply:
that looks like you're jumping through a lot of hoops just to get back to what I've written here.